sympy/polys/matrices/linsolve.py

```diff
@@ -126,53 +129,70 @@ def sympy_dict_to_dm(eqs_coeffs, eqs_rhs, syms):
     for eq, rhs in zip(eqs_coeffs, eqs_rhs):
         eqdict = {sym2index[s]: elem_map[c] for s, c in eq.items()}
         if rhs:
-            eqdict[nsyms] = - elem_map[rhs]
+            eqdict[nsyms] = -elem_map[rhs]
         if eqdict:
             eqsdict.append(eqdict)
-    sdm_aug = SDM(enumerate(eqsdict), (neqs, nsyms+1), K)
+    sdm_aug = SDM(enumerate(eqsdict), (neqs, nsyms + 1), K)
     return sdm_aug
 
 
-def _expand_eqs_deprecated(eqs):
-    """Use expand to cancel nonlinear terms.
+def _linear_eq_to_dict(eqs, syms):
+    """Convert a system Expr/Eq equations into dict form, returning
+    the coefficient dictionaries and a list of syms-independent terms
+    from each expression in ``eqs```.
 
-    This approach matches previous behaviour of linsolve but should be
-    deprecated.
+    Examples
+    ========
+
+    >>> from sympy.polys.matrices.linsolve import _linear_eq_to_dict
+    >>> from sympy.abc import x
+    >>> _linear_eq_to_dict([2*x + 3], {x})
+    ([{x: 2}], [3])
     """
-    def expand_eq(eq):
-        if eq.is_Equality:
-            eq = eq.lhs - eq.rhs
-        return eq.expand()
+    coeffs = []
+    ind = []
+    symset = set(syms)
+    for i, e in enumerate(eqs):
+        if e.is_Equality:
+            (coeff, terms), (cR, tR) = [_lin_eq2dict(ai, symset)
+                for ai in e.args]
```

Comment:
A loop to avoid repeating a line just twice is not really a good application of DRY. It's a lot easier to understand the straight-forward repetitive code:
```python
coeff, terms = _lin_eq2dict(e.lhs, symset)
cR, tR = _lin_eq2dict(e.rhs, symset)
```